src/app/games/hive/hive.component.ts

```diff
@@ -0,0 +1,298 @@
+        const boardAndRemainingViewBox: ViewBox = this.boardViewBox
+            .containingAtLeast(minimalViewBox)
+            .expand(0, 0, spaceForRemainingPieces, spaceForRemainingPieces);
+        const inspectedStackPosition: Coord =
+            new Coord(boardAndRemainingViewBox.right() + this.SPACE_SIZE,
+                      boardAndRemainingViewBox.center().y);
+        this.inspectedStackTransform = `translate(${inspectedStackPosition.x} ${inspectedStackPosition.y})`;
+
+        const spaceForInspectedStack: number = this.SPACE_SIZE*5;
+        this.viewBox = boardAndRemainingViewBox
+            .expand(0, spaceForInspectedStack, 0, 0)
+            .toSVGString();
+    }
+
+    private getAllNeighbors(): Coord[] {
+        const neighbors: MGPSet<Coord> = new MGPSet();
+        for (const piece of this.pieces) {
+            neighbors.union(new MGPSet(this.getState().emptyNeighbors(piece.coord)));
+        }
+        if (neighbors.isEmpty()) {
+            // We need at least one clickable coord to be playable at first turn
+            neighbors.add(new Coord(0, 0));
+        }
+        return neighbors.toList();
+    }
+
+    public cancelMoveAttempt(): void {
+        this.selectedRemaining = MGPOptional.empty();
+        this.selectedStart = MGPOptional.empty();
+        this.selectedSpiderCoords = [];
+        this.selected = [];
+        this.indicators = [];
+        this.inspectedStack = MGPOptional.empty();
+    }
+
+    public getRemainingPieceTransformAsCoord(piece: HivePiece): Coord {
+        const shift: number = this.getRemainingPieceShift(piece);
+        const x: number = this.boardViewBox.center().x + shift * this.SPACE_SIZE * 4;
+        let y: number;
+        if (piece.owner === this.role) {
+            // Current player is below
+            y = this.boardViewBox.bottom() + (this.SPACE_SIZE * 3);
+        } else {
+            y = this.boardViewBox.up - (this.SPACE_SIZE * 2);
+        }
+        return new Coord(x, y);
+    }
+
+    public getRemainingPieceTransform(piece: HivePiece): string {
+        const transform: Coord = this.getRemainingPieceTransformAsCoord(piece);
+        return `translate(${transform.x} ${transform.y})`;
+    }
+
+    public getRemainingPieceHighlightTransform(piece: HivePiece): string {
+        const transform: Coord = this.getRemainingPieceTransformAsCoord(piece);
+        const size: number = this.getState().remainingPieces.getRemaining(piece);
+        return `translate(${transform.x} ${transform.y - (this.PIECE_HEIGHT * size)})`;
+    }
+
+    private getRemainingPieceShift(piece: HivePiece): number {
+        switch (piece.kind) {
+            case 'QueenBee': return -2.5;
+            case 'Beetle': return -1.5;
+            case 'Grasshopper': return -0.5;
+            case 'Spider': return 0.5;
+            default:
+                Utils.expectToBe(piece.kind, 'SoldierAnt', 'piece must be a soldier ant');
```

Comment:
but is what :p ?

Reply:
oooh mais c'est expectToBe qui devrait se charger de ça, pas nos appels !!

Reply:
ah non car on veut être flexible et ça ça va nous faire des messages d'erreurs pas clairs !

Reply:
Ici si j'enlève le message, expectToBe ferait par défaut : `"A default switch case did not observe the correct value, expected SoldierAnt but got value Blibli instead", donc je fais ça !